### PR TITLE
Update all dependencies manually

### PR DESCRIPTION
**Pull request by dougbu**
- add `CoherentParentDependency` attributes for the new `darc` version
- also update docs/ReferenceResolution.md to match aspnet/Extensions version (and pick up improvements there)

### CODE REVIEW

**Review by pranavkm — APPROVED** [review]
As long as it works

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -9,384 +9,384 @@
...
+      <Sha>ef145001b10660be940aeb9318e0ff7e306f65bc</Sha>
     </Dependency>
-    <Dependency Name="System.Threading.Channels" Version="4.6.0-preview4.19157.3">
+    <Dependency Name="System.Threading.Channels" Version="4.6.0-preview4.19162.3" CoherentParentDependency="Microsoft.NETCore.App">
       <Uri>https://github.com/dotnet/corefx</Uri>
-      <Sha>00000</Sha>
+      <Sha>ef145001b10660be940aeb9318e0ff7e306f65bc</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Extensions.DependencyModel" Version="3.0.0-preview4-27507-10">
+    <Dependency Name="Microsoft.Extensions.DependencyModel" Version="3.0.0-preview4-27512-06" CoherentParentDependency="Microsoft.NETCore.App">
```
Comment (dougbu) [inline eng/Version.Details.xml]: @mmitche as in aspnet/EntityFrameworkCore#14999, does this `CoherentParentDependency` attribute hurt?

Comment (mmitche) [reply eng/Version.Details.xml]: Nope will be ignored until supported.

**Comment by dougbu** [comment]
@ajcvickers looks like 8 tests are failing for similar reasons to what we saw in the `release/3.0-preview3` branch e.g. missing a `IMemoryCache` implementation in DI. Is this what you were expecting?

Affected tests include
1. `MusicStore.Controllers.CheckoutControllerTest.AddressAndPayment_RedirectToCompleteWhenSuccessful`
2. `MusicStore.Controllers.CheckoutControllerTest.Complete_ReturnsErrorIfInvalidOrder`
3. `MusicStore.Controllers.ShoppingCartControllerTest.RemoveFromCart_RemovesItemFromCart`
4. `MusicStore.Controllers.CheckoutControllerTest.Complete_ReturnsOrderIdIfValid`
5. `MusicStore.Controllers.StoreControllerTest.Details_ReturnsHttpNotFoundWhenNoAlbumData`
    - missing `IMemoryCache` implementation
6. `MusicStore.Controllers.StoreControllerTest.Details_ReturnsAlbumDetail`
    - missing `IMemoryCache` implementation
7. `MusicStore.Controllers.ShoppingCartControllerTest.AddToCart_AddsItemToCart
8. `MusicStore.Controllers.HomeControllerTest.Index_GetsSixTopAlbums`
    - missing `IMemoryCache` implementation

All are hitting NREs in `Log(...)` unless otherwise marked.

**Comment by dougbu** [comment]
All `AspNetCore-ci` jobs are now complete. The 8 failures were consistent across the `Test: *` jobs.

**Comment by dougbu** [comment]
Once we merge #8478, this should be good to go. I'll rebase and builds should succeed 😺

**Comment by ajcvickers** [comment]
@dougbu Merged my changes.

**Comment by dougbu** [comment]
@ajcvickers thanx again. I've rebased. Now it's up to the CI checks&hellip;

**Pull request merged**